docs/Additional_Build_Documentation.md

```diff
@@ -0,0 +1,296 @@
+# Building Benchmark
+
+To build Benchmark we need to follow a series of simple steps:
+*   Obtain Benchmark
+*   Configure Benchmark
+*   Build Benchmark
+
+To obtain Benchmark we can simply clone the Github repository using
+```bash
+$ git clone https://github.com/google/benchmark.git
+```
+Now, Benchmark uses Google Test to run it's tests. So, if we wish to build and run
+the tests we need to include googletest, which can be done either by cloning the googletest
+repository or by asking benchmark in the configuration step to download this dependency.
+We can clone the googletest repository in benchmark using
+```bash
+$ git clone https://github.com/google/googletest.git benchmark/googletest
+```
+To configure Benchmark we use CMake as follows:
+```bash
+$ cmake -G <generator> [options] -S benchmark -B build
+```
+If we haven't cloned the googletest repository as described above we can ask benchmark to download this dependency in the above configuration step by adding `-DBENCHMARK_DOWNLOAD_DEPENDENCIES=ON`
+
+Then to build Benchmark we use CMake again as follows:
+```bash
+$ cmake -build build
+```
+If we don't want the Benchmark tests to build and run we can add `-DBENCHMARK_ENABLE_GTEST_TESTS=OFF` in the configuration step.
+
+## Using Benchmark
+
+If we want to use Benchmark with a particular file, say `BenchmarkEverything.cpp`,
+we can build Benchmark and use the following comands in the terminal.
+
+### In Windows
+```bash
+> cl /EHsc /MD BenchmarkEverything.cpp /I<BENCHMARK_SOURCE_DIR>\include /link shlwapi.lib <BENCHMARK_BUILD_DIR>\src\Release\benchmark.lib
```

Comment:
i presume there's an easier way if you're using VS

Reply:
yes, others will have syntax similar to linux(I think)